src/components/ModalDialog/ModalDialog.vue

```diff
@@ -0,0 +1,282 @@
+</template>
+
+<script>
+import UnnnicIcon from '../Icon.vue';
+import UnnnicButton from '../Button/Button.vue';
+import UnnnicI18n from '../../mixins/i18n';
+
+export default {
+  name: 'UnnnicModalDialog',
+  components: {
+    UnnnicIcon,
+    UnnnicButton,
+  },
+  mixins: [UnnnicI18n],
+  props: {
+    modelValue: {
+      type: Boolean,
+      required: true,
+    },
+    persistent: {
+      type: Boolean,
+      default: false,
+    },
+    type: {
+      type: String,
+      default: '',
+      validate(type) {
+        return ['success', 'warning', 'attention'].includes(type);
+      },
+    },
+    size: {
+      type: String,
+      default: 'md',
+      validate(size) {
+        return ['sm', 'md', 'lg'].includes(size);
+      },
+    },
+    title: {
+      type: String,
+      default: 'title',
+    },
+    icon: {
+      type: String,
+      default: '',
+    },
+    iconScheme: {
+      type: String,
+      default: '',
+    },
+    showCloseIcon: {
+      type: Boolean,
+      default: false,
+    },
+    primaryButtonText: {
+      type: String,
+      default: '',
+    },
+    secondaryButtonText: {
+      type: String,
+      default: '',
+    },
+    primaryButtonType: {
+      type: String,
+      default: '',
+    },
+    showActionsDivider: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  emits: ['primaryButtonClick', 'secondaryButtonClick', 'update:modelValue'],
+
+  data() {
+    return {
+      defaultTranslations: {
+        cancel: {
+          'pt-br': 'Cancelar',
+          en: 'Cancel',
+          es: 'Cancelar',
+        },
+      },
+      iconsMapper: {
+        success: { icon: 'check_circle', scheme: 'aux-green-500' },
+        warning: { icon: 'warning', scheme: 'aux-red-500' },
+        attention: { icon: 'error', scheme: 'aux-yellow-500' },
+      },
+      primaryButtonTypeMapper: {
+        success: 'primary',
+        warning: 'warning',
+        attention: 'attention',
+      },
+    };
+  },
+  watch: {
+    modelValue(value) {
+      if (value) document.body.style.overflow = 'hidden';
+      else document.body.style.overflow = '';
+    },
+  },
+  methods: {
+    close() {
+      this.$emit('update:modelValue', false);
+    },
+  },
+};
+</script>
+
+<style lang="scss" scoped>
+@import '../../assets/scss/unnnic.scss';
+* {
+  margin: 0;
+  padding: 0;
+  box-sizing: border-box;
+}
+.unnnic-modal-dialog {
+  width: 100vw;
+  height: 100vh;
+  position: fixed;
+  left: 0;
+  top: 0;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  z-index: 9999;
+
+  &__overlay {
+    position: absolute;
+    width: 100%;
+    height: 100%;
+    background-color: rgba(0, 0, 0, 0.4);
+  }
+}
+
+.unnnic-modal-dialog__container {
+  display: flex;
+  flex-direction: column;
+  background: $unnnic-color-neutral-white;
+  border-radius: $unnnic-spacing-xs;
+  box-shadow: 0px 4px 8px 0px rgba(0, 0, 0, 0.1);
+  position: fixed;
+  max-height: calc(100vh - $unnnic-spacing-md * 2);
```

Comment:
Why you preffer use `$unnnic-spacing-md * 2` instead only `$unnnic-spacing-giant`?